### ISSUE TITLE
Regex search for newlines

### ISSUE THREAD

**Issue by oonet** [tsujan/FeatherPad#421]
Regex doesn't see \n (but does see newline characters pasted into search string).

**Comment by tsujan** [comment]
Yes, you're right and I'm aware of it. Unfortunately, I found no *simple* way to connect how Qt processes blocks (=lines) with how regex handles newline. Actually, I devised a method to support newlines in regex search and it worked but it was so complicated that I didn't add it to the code.

I leave this page open. Who knows? Maybe an elegant solution exits or the above-mentioned method can be simplified and added to the code.

**Comment by oonet** [comment]
Not ideal, but maybe substitute newline character (as pasted) for all occurrences of \n in search string?

**Comment by tsujan** [comment]
Regex can include things like `(?=\n)`, `(?<!\n)` or `(\s\n){1,3}`. I can just hope that the problem only *seems* so complex and a simple solution exists somehow...

**Comment by oonet** [comment]
I feel for you!

Just testing, it appears FeatherPad finds newline characters only with regex unselected.

Is there a multiline flag that must be set?

**Comment by tsujan** [comment]
> it appears FeatherPad finds newline characters only with regex unselected.

I added my reply to a wrong page ;) See https://github.com/tsujan/FeatherPad/issues/418#issuecomment-697943662
